Snowflake.Data/Core/SFSessionProperty.cs

```diff
@@ -50,6 +50,35 @@ class SFSessionProperties : Dictionary<SFSessionProperty, String>
     {
         static private SFLogger logger = SFLoggerFactory.GetLogger<SFSessionProperties>();
 
+        public override bool Equals(object obj)
+        {
+            if (obj == null) return false;
+            try
+            {
+                SFSessionProperties prop = (SFSessionProperties)obj;

```

Comment:
do you check the number of properties?

Reply:
line 61 would check every property should either both exist or both not exist.